test/wallet-migration-test.js

```diff
@@ -540,6 +571,126 @@ describe('Wallet Migrations', function() {
+    };
+
+    let walletDB, ldb;
+    before(async () => {
+      WalletMigrator.migrations = {};
+      await fs.mkdirp(location);
+    });
+
+    after(async () => {
+      WalletMigrator.migrations = migrationsBAK;
+      await rimraf(location);
+    });
+
+    beforeEach(async () => {
+      walletDB = new WalletDB(walletOptions);
+      ldb = walletDB.db;
+    });
+
+    afterEach(async () => {
+      if (ldb.opened)
+        await ldb.close();
+    });
+
+    const newToOld = (raw) => {
+      const preLen = 1 + 1 + 1 + 1 + 4 + 4;
+      const pre = raw.slice(0, preLen);
+      const lookahead = raw.slice(preLen, preLen + 4);
+      const post = raw.slice(preLen + 4);
+
+      return Buffer.concat([
+        pre,
+        Buffer.alloc(1, lookahead[0]),
+        post
+      ]);
+    };
+
+    it('should write old account record', async () => {
+      const setupLookahead = async (wallet, n) => {
+        const accounts = [];
+        let b;
+
+        for (let i = 0; i < 3; i++)
+          accounts.push(await wallet.getAccount(i));
+
+        b = ldb.batch();
+        for (let i = 0; i < 3; i++)
+          await accounts[i].setLookahead(b, n + i);
+        await b.write();
+
+        b = ldb.batch();
+        for (let i = 0; i < 3; i++) {
+          const encoded = newToOld(accounts[i].encode(), n + i);
+          b.put(layout.a.encode(wallet.wid, i), encoded);
+        }
+        await b.write();
+      };
+
+      await walletDB.open();
+
+      const wallet = walletDB.primary;
+      await wallet.createAccount({});
+      await wallet.createAccount({});
+
+      await setupLookahead(wallet, TEST_LOOKAHEAD + 0);
+
+      const wallet2 = await walletDB.create({});
+      await wallet2.createAccount({});
+      await wallet2.createAccount({});
+      await setupLookahead(wallet2, TEST_LOOKAHEAD + 10);
+
+      await walletDB.close();
+    });
+
+    it('should enable wallet account lookahead migration', () => {
+      WalletMigrator.migrations = {
+        0: WalletMigrator.MigrateAccountLookahead
+      };
+    });
+
+    it('should fail without migrate flag', async () => {
+      const expectedError = migrationError(WalletMigrator.migrations, [0],
+        wdbFlagError(0));
+
+      await assert.rejects(async () => {
+        await walletDB.open();
+      }, {
+        message: expectedError
+      });
+
+      await ldb.close();
+    });
+
+    it('should migrate with migrate flag', async () => {
+      const checkLookahead = async (wallet, n) => {
+        for (let i = 0; i < 3; i++) {
+          const account = await wallet.getAccount(i);
+          assert.strictEqual(account.lookahead, n + i);
+        }
+      };
+
+      walletDB.options.walletMigrate = 0;
+
+      await walletDB.open();
+      const wallet = walletDB.primary;
+      const wallet2 = await walletDB.get(1);
+      await checkLookahead(wallet, TEST_LOOKAHEAD + 0);
+      await checkLookahead(wallet2, TEST_LOOKAHEAD + 10);
+      await walletDB.close();
```

Comment:
now that migration is complete, we could add one more test setting lookahead `> 255`, close and re-open DB and make sure the new value is there and not rolled over a U8

Reply:
This makes sure migration is correct, that test is wallet-unit-test. (Also because of the new format that test has to manually convert new to old)